### PR TITLE
fix: use qualifyColumn rather than assuming format

### PR DESCRIPTION
**Pull request by willtj**
In my setup I have some Laravel models stored in MongoDB and others in MySQL. MorphTo and BelongsTo relations aren't loading where the parent model is in MongoDB and the child is in MySQL, i.e:

```
class Parent extends MongoModel
{

}

class Child extends EloquentModel
{
    public function myRelation()
    {
        return $this->belongsTo(Parent::class);
    }
}
```

`Child::first()->myRelation` will return `null` because `addConstraints()` is assuming that the key name should be prepended with the table name, which isn't the case for MongoDB.

The `getQualifiedOwnerKeyName()` method already exists on the BelongsTo class, is there any reason why it's not currently being used when adding constraints?

### CODE REVIEW

File: tests/Database/DatabaseEloquentMorphToTest.php [hunk 1]
```diff
@@ -378,8 +379,9 @@ public function getRelation($parent = null, $builder = null)
         $this->builder = $builder ?: m::mock(Builder::class);
         $this->builder->shouldReceive('where')->with('relation.id', '=', 'foreign.value');
         $this->related = m::mock(Model::class);
-        $this->related->shouldReceive('getKeyName')->andReturn('id');
+        $this->related->shouldReceive('getcolumn')->andReturn('id');
```
Comment (AndrewMast) [inline tests/Database/DatabaseEloquentMorphToTest.php:382]: Is this supposed to be `getColumn`?

Comment (GrahamCampbell) [reply tests/Database/DatabaseEloquentMorphToTest.php:382]: doesn't really matter. method names are not case sensitive.

**Comment by crynobone** [comment]
This PR broken `morphTo()` usage. If for example we explicitly add `qualifyColumn(string $column)` multiple tests will be broken indicating an issue.

**Comment by crynobone** [comment]
This PR has been reverted due as it triggers error on certain `MorphTo` relationships.  See https://github.com/laravel/framework/pull/53568

**Pull request merged**